### ISSUE TITLE
Rule proposal: `empty-brace-space`

### ISSUE THREAD

**Issue by Richienb** [sindresorhus/eslint-plugin-unicorn#900]
This rule would enforce spaces between braces in empty anonymous functions and try/catch braces. It is important because it is a syntax style I [need to follow](https://github.com/sindresorhus/internal-ip/pull/36#discussion_r527539515) but isn't automatically caught by xo already.

## Fail

```js
try { } catch { }

() => { }
```

## Pass

```js
try {} catch {}

() => {}
```

**Comment by fisker** [comment]
```js
for (;;) {}
```

Or

```js
for (;;);
```

?

**Comment by sindresorhus** [comment]
@fisker The rule would ignore the second case as it doesn't have any braces.

**Comment by fisker** [comment]
I mean

If it's

```js
for (;;) { }
```

Should we fix to 

```js
for (;;) {}
```

Or

```js
for (;;);
```

?

**Comment by Richienb** [comment]
Since an empty for loop is useless, the fix is removing it completely.

**Comment by sindresorhus** [comment]
To `for (;;) {}`. This rule should only be about "brace space". Nothing else.

**Comment by fisker** [comment]
Should we fix `a = { }` and `class A { }`, they are different AST type with cases above.

**Comment by sindresorhus** [comment]
@fisker Yes, anywhere `{} ` can be used. Anonymous classes too: `const a = class {}`

**Comment by sindresorhus** [comment]
This is now accepted. PR welcome.

**Comment by fisker** [comment]
I think this rule is easy to implement, problem is how many cases? 

 In ast empty block in  `for` `while` `do-while` `array function`   `try/catch/finnal` the same, and two types metioned above, anything else?

---
Cases we should ignore

- `const {} = a`
- `import {} from "a"`

**Comment by fisker** [comment]
Ha, here https://github.com/eslint/eslint/blob/06b58096975935ec016d96dd5f333f059c270f26/lib/rules/brace-style.js#L154

**Comment by fisker** [comment]
I'm going to implement this now.

**Comment by ksnyde** [comment]
i get this error but when I correct it, vs-code automatically injects a space back in when I save. Anyone know what rule is doing this?
